src/mono/wasm/testassets/WasmBasicTestApp/wwwroot/main.js

```diff
@@ -21,6 +25,37 @@ switch (testCase) {
     case "AppSettingsTest":
         dotnet.withApplicationEnvironment(params.get("applicationEnvironment"));
         break;
+    case "DownloadResourceProgressTest":
+        if (params.get("failAssemblyDownload") === "true") {
+            let assemblyCounter = 0;
+            let failAtAssemblyNumbers = [
+                Math.floor(Math.random() * 5),
+                Math.floor(Math.random() * 5) + 5,
+                Math.floor(Math.random() * 5) + 10
+            ];
+            dotnet.withDiagnosticTracing(true).withResourceLoader((type, name, defaultUri, integrity) => {
+                if (type !== "assembly")
+                    return defaultUri;
+
+                assemblyCounter++;
+                if (!failAtAssemblyNumbers.includes(assemblyCounter))
+                    return defaultUri;
+
+                testOutput("Throw error instead of downloading resource");
+                const error = new Error("Simulating a failed fetch");
+                error.silent = true;
+                throw error;
```

Comment:
Does this get surfaced to the user at all? What behavior does blazor have when we fail to load an assembly at all?

Reply:
Before this PR, the Blazor aborts on first download failure. Their original approach was to let the user override `loadBootResource` do retries by themself. After discussion with @pavelsavara we agreed that turn-on our built-in retry support.

Reply:
I meant, is this surfaced when even a 3rd retry fails?